core/src/main/scala-3/sttp/tapir/internal/EnumMacros.scala

```diff
@@ -0,0 +1,44 @@
+package sttp.tapir.internal
+
+import scala.quoted.*
+
+private[tapir] object EnumMacros:
+
+  transparent inline def isScalaEnum[T]: Boolean = inline compiletime.erasedValue[T] match
```

Comment:
This name can be confusing, because this checks for more than a simple scala `enum`. 

Maybe we could assume the following convention, using the name "`enum`" for all cases of Scala 3 `enum`s (with or without parameters). On the other hand "`enumeration`" would mean "true" enumerations, that is parameterless `enum`s or sealed traits with only case objects as children. It's not perfect as one is an abbrievation of the other, but I think if used consistently (we would probably  need to document it in some developer docs - maybe readme?) it would be clear what we mean.

Reply:
Makes sense, I propose 'contributing.md' for such documentation. It also looks like a good place for versioning scheme https://github.com/softwaremill/tapir/issues/3111, wdyt?